### ISSUE TITLE
Building amazon instance creates an AMI, but errors

### ISSUE THREAD

**Issue by jsheflin** [mitchellh/packer#4360]
Building amazon instance creates an AMI, but errors:

2017/01/04 15:25:06 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/01/04 15:25:06 machine readable: amazon-instance,error []string{"unexpected EOF"}
2017/01/04 15:25:06 ui error: --> amazon-instance: unexpected EOF

-Packer v0.12.1
-Ubuntu 16
- https://gist.github.com/jsheflin/6fe795727b3fac4c82eaaea46ad695c2

**Comment by joshuaspence** [comment]
I am also seeing this problem

**Comment by mwhooker** [comment]
thanks for the report. Here's the interesting bit

```
2017/01/04 17:06:00 packer: panic: interface conversion: interface {} is nil, not map[string][]string
2017/01/04 17:06:00 packer: 
2017/01/04 17:06:00 packer: goroutine 34 [running]:
2017/01/04 17:06:00 packer: panic(0x13e6f60, 0xc42058d380)
2017/01/04 17:06:00 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
2017/01/04 17:06:00 packer: github.com/mitchellh/packer/builder/amazon/common.(*StepAMIRegionCopy).Run(0xc420344ed0, 0x1e117a0, 0xc420344e40, 0x0)
2017/01/04 17:06:00 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/builder/amazon/common/step_ami_region_copy.go:26 +0x8ff
2017/01/04 17:06:00 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420322ec0, 0x1e117a0, 0xc420344e40)
2017/01/04 17:06:00 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x207
2017/01/04 17:06:00 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*DebugRunner).Run(0xc420344f00, 0x1e117a0, 0xc420344e40)
2017/01/04 17:06:00 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/debug_runner.go:67 +0x335
2017/01/04 17:06:00 packer: github.com/mitchellh/packer/builder/amazon/instance.(*Builder).Run(0xc420268000, 0x1e140e0, 0xc4203503c0, 0x1e0d0a0, 0xc42034ee60, 0x1e12e60, 0xc420144178, 0xc42024c660, 0x0, 0x0, ...)
2017/01/04 17:06:00 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/builder/amazon/instance/builder.go:277 +0x1039
2017/01/04 17:06:00 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc420146c00, 0x1, 0xc42024e2b0, 0x0, 0x0)
2017/01/04 17:06:00 packer: 	/Users/mwhooker/go/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x218
2017/01/04 17:06:00 packer: reflect.Value.call(0xc420286840, 0xc4201440b0, 0x13, 0x15ffd31, 0x4, 0xc4200d7ed0, 0x3, 0x3, 0x0, 0x0, ...)
2017/01/04 17:06:00 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:434 +0x5c8
2017/01/04 17:06:00 packer: reflect.Value.Call(0xc420286840, 0xc4201440b0, 0x13, 0xc4200d7ed0, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/01/04 17:06:00 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:302 +0xa4
2017/01/04 17:06:00 packer: net/rpc.(*service).call(0xc420148880, 0xc420148840, 0xc420142688, 0xc42014a780, 0xc420146ea0, 0x13117c0, 0xc42024e27c, 0x18a, 0x12e6d00, 0xc42024e2b0, ...)
2017/01/04 17:06:00 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:383 +0x148
2017/01/04 17:06:00 packer: created by net/rpc.(*Server).ServeCodec
2017/01/04 17:06:00 packer: 	/usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:477 +0x421
2017/01/04 17:06:00 ui error: [1;31mBuild 'amazon-instance' errored: unexpected EOF[0m
```

will take a look tomorrow

**Comment by DanHam** [comment]
@mwhooker I could be wrong, but this looks like a duplicate of #4340. If so it's been fixed by #4341.

**Comment by mwhooker** [comment]
@DanHam you're very right, thanks!

@jsheflin this should be fixed in master and will be in the next release

**Comment by joshuaspence** [comment]
I don't think this this is fixed. Yesterday I saw this failure both on the latest release and building from `master`

**Comment by mwhooker** [comment]
@joshuaspence hmm, interesting. Could you please post a crash log when building from master?

**Comment by joshuaspence** [comment]
Here is the crash log when running from `master`:

```
panic: interface conversion: interface {} is nil, not map[string][]string
2017/01/06 08:14:31 packer: 
2017/01/06 08:14:31 packer: goroutine 27 [running]:
2017/01/06 08:14:31 packer: panic(0x192b840, 0xc82030fb00)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/builder/amazon/common.(*StepAMIRegionCopy).Run(0xc8204348d0, 0x7fcbf02b4580, 0xc820434840, 0x0)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/builder/amazon/common/step_ami_region_copy.go:26 +0xab2
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc8204365c0, 0x7fcbf02b4580, 0xc820434840)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x330
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/builder/amazon/instance.(*Builder).Run(0xc820220580, 0x7fcbf02b3fd8, 0xc820402400, 0x7fcbf02b4020, 0xc820408810, 0x7fcbf02b4050, 0xc8202cc268, 0x0, 0x0, 0x0, ...)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/builder/amazon/instance/builder.go:277 +0x1bba
2017/01/06 08:14:31 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc8202824c0, 0x1, 0xc820201be0, 0x0, 0x0)
2017/01/06 08:14:31 packer:     /REDACTED/gopath/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x3b9
2017/01/06 08:14:31 packer: reflect.Value.call(0x17b6500, 0x19a6a18, 0x13, 0x1c54028, 0x4, 0xc8205b9ec8, 0x3, 0x3, 0x0, 0x0, ...)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/reflect/value.go:435 +0x120d
2017/01/06 08:14:31 packer: reflect.Value.Call(0x17b6500, 0x19a6a18, 0x13, 0xc8205b9ec8, 0x3, 0x3, 0x0, 0x0, 0x0)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/reflect/value.go:303 +0xb1
2017/01/06 08:14:31 packer: net/rpc.(*service).call(0xc8202940c0, 0xc820294080, 0xc820296090, 0xc820151880, 0xc8202b0560, 0x156f760, 0xc820201bdc, 0x18a, 0x152f140, 0xc820201be0, ...)
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/net/rpc/server.go:383 +0x1c2
2017/01/06 08:14:31 packer: created by net/rpc.(*Server).ServeCodec
2017/01/06 08:14:31 packer:     /usr/lib/go-1.6/src/net/rpc/server.go:477 +0x49d
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
2017/01/06 08:14:31 ui error: Build 'hvm.instance-store' errored: unexpected EOF
2017/01/06 08:14:31 Builds completed. Waiting on interrupt barrier...
2017/01/06 08:14:31 machine readable: error-count []string{"1"}
2017/01/06 08:14:31 ui error: 
==> Some builds didn't complete successfully and had errors:
2017/01/06 08:14:31 machine readable: hvm.instance-store,error []string{"unexpected EOF"}
2017/01/06 08:14:31 ui error: --> hvm.instance-store: unexpected EOF
2017/01/06 08:14:31 ui: 
==> Builds finished but no artifacts were created.
2017/01/06 08:14:31 waiting for all plugin processes to complete...
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
2017/01/06 08:14:31 /REDACTED/gopath/bin/packer: plugin process exited
```
